ee/hogai/eval/ci/eval_memory_onboarding.py

```diff
@@ -0,0 +1,397 @@
+- pass: Follows all required formatting specifications exactly
+- fail: Missing required formatting elements or includes prohibited content""",
+            choice_scores={
+                "pass": 1.0,
+                "fail": 0.0,
+            },
+            model="gpt-4.1",
+            **kwargs,
+        )
+
+
+@pytest.fixture(scope="module")
+def core_memory():
+    """We override the conftest.py core_memory fixture, which has autouse=True, as for memory onboarding a CoreMemory must be ABSENT."""
+    pass
+
+
+@pytest.fixture
+def call_memory_onboarding():
+    """Fixture to call MemoryOnboardingNode with a parametrized EventTaxonomyItem as input."""
+
+    async def callable(input: tuple[str, Optional[EventTaxonomyItem]]) -> Optional[AssistantMessage]:
+        organization_name, taxonomy_item = input  # The org name MAY matter as it gets included in the system prompt
+        # Because memory is stateful at the team level, each eval case here must have its own team instance
+        _, team, user = await database_sync_to_async(User.objects.bootstrap)(
+            organization_name=organization_name, email=f"{uuid4()}@example.com", password=None
+        )
+        conversation = await Conversation.objects.acreate(team=team, user=user)
+
+        graph = (
+            AssistantGraph(team=team, user=user)
+            .add_memory_onboarding(AssistantNodeName.END)
+            .compile(checkpointer=False)
+        )
+        raw_state = await graph.ainvoke(
+            AssistantState(messages=[HumanMessage(content=SLASH_COMMAND_INIT)]),
+            # Mock the _retrieve_context method to return our input data via `configurable`
+            # Because evals run concurrently and everything's async, patch() just doesn't work
+            {"configurable": {"thread_id": conversation.id, "_mock_memory_onboarding_context": taxonomy_item}},
+        )
+
+        state = AssistantState.model_validate(raw_state)
+        if not state.messages:
+            return None
+        last_message = state.messages[-1]
+        if not isinstance(last_message, AssistantMessage):
+            return None
+        return last_message
+
+    return callable
+
+
+@pytest.mark.django_db
+async def eval_memory_onboarding(call_memory_onboarding, pytestconfig):
+    await MaxPublicEval(
+        experiment_name="memory_onboarding",
+        task=call_memory_onboarding,
+        scores=[
+            SatisfiesProductDetails(),
+            SatisfiesBusinessDetails(),
+            SatisfiesTechnicalDetails(),
+            HasCorrectStyle(),
+        ],
+        data=[
+            # Case: SaaS, well-known and massive in scope (so this one is tricky to describe in any detail)
+            EvalCase(
+                input=(
+                    "Salesforce",
+                    EventTaxonomyItem(property="$host", sample_values=["salesforce.com"], sample_count=1),
+                ),
+                expected="""
```

Comment:
Not sure how `expected` is used in scoring for the first 3 cases, could you clarify, please? Is it cleverly used in the scorers above?

I assumed we need `expected` only for the routing checks that rely on it (`ENQUIRY_INITIAL_MESSAGE` and `SCRAPING_TERMINATION_MESSAGE`)

Reply:
Well, we actually didn't use it for scoring. I used it mostly for reference. But let's actually put those in as examples for the scorers.